src/Integration.Vsix/Suppression/SuppressionManager.cs

```diff
@@ -31,24 +31,25 @@ namespace SonarLint.VisualStudio.Integration.Vsix.Suppression
     internal sealed class SuppressionManager : IDisposable
     {
         private readonly IServiceProvider serviceProvider;
-        private readonly ITimerFactory timerFactory;
         private readonly IActiveSolutionBoundTracker activeSolutionBoundTracker;
-        private readonly ISonarQubeService sonarQubeService;
         private readonly ISonarLintOutput sonarLintOutput;
 
         private DelegateInjector delegateInjector;
         private ISonarQubeIssuesProvider sonarqubeIssueProvider;
         private SuppressionHandler suppressionHandler;
 
-        public SuppressionManager(IServiceProvider serviceProvider, ITimerFactory timerFactory)
+        public SuppressionManager(IServiceProvider serviceProvider)
         {
+            if (serviceProvider == null)
+            {
+                throw new ArgumentNullException(nameof(serviceProvider));
+            }
+
             this.serviceProvider = serviceProvider;
-            this.timerFactory = timerFactory;
 
             this.activeSolutionBoundTracker = serviceProvider.GetMefService<IActiveSolutionBoundTracker>();
             this.activeSolutionBoundTracker.SolutionBindingChanged += OnSolutionBindingChanged;
 
-            this.sonarQubeService = serviceProvider.GetMefService<ISonarQubeService>();
```

Comment:
I made a couple of changes to make the _SuppressionManager_ easier to test:
1) removed the _TimerFactory_ from the constructor. The SM didn't call it - it just passed it on in the _SetupSuppressionHandling_ method.
2) moved the call to fetch the _SonarQubeService_  to the _SetupSuppressionHandling_ method. It isn't used anywhere else, and there are already a couple of _GetService_ calls in that method.